usecases/scheduled_execution/batch_filtering.go

```diff
@@ -0,0 +1,185 @@
+package scheduled_execution
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/models/ast"
+	"github.com/checkmarble/marble-backend/pure_utils/duration"
+
+	"github.com/jackc/pgx/v5"
+)
+
+func selectFiltersFromTriggerAstRootAnd(node ast.Node, table models.TableIdentifier) []models.Filter {
```

Comment:
Good anticipation to the possibility of OR trigger condition

Reply:
haha yea, for this one I actually took inspiration from the front end !